sklearn/decomposition/_fastica.py

```diff
@@ -588,10 +595,12 @@ def fit_transform(self, X, y=None):
             and n_features is the number of features.
 
         y : Ignored
+            Not used, present for API consistency by convention.
 
         Returns
         -------
-        X_new : ndarray of shape (n_samples, n_components)
+        S : ndarray of shape (n_samples, n_components)
+            Sources matrix.
```

Comment:
Since this is the same data as the `transform` method which is public I would rather keep the `X_new` name and the same description of the returned value of that method for consistency.

The `_fit` method is private, so won't appear in the online API doc, so it's not a problem if the local variable S does not have the same name here.